django_auth_ldap/backend.py

```diff
@@ -537,11 +557,20 @@ def _search_for_user():
                     "AUTH_LDAP_USER_SEARCH must be an LDAPSearch instance."
                 )
 
-            results = search.execute(self.connection, {"user": self._username})
-            if results is not None and len(results) == 1:
-                (user_dn, self._user_attrs) = next(iter(results))
-            else:
-                user_dn = None
+            user_dn = None
+
+            try:
+                results = search.execute(self.connection, {"user": self._username})
+                if results is not None and len(results) == 1:
+                    (user_dn, self._user_attrs) = next(iter(results))
```

Comment:
Since `execute` already iterated over the results in `_process_results`, I think this block can be taken out of the `try` and moved to an `else`, as it shouldn’t raise an `LDAPError` ?